### PR TITLE
Feature/finish task modal

### PR DESCRIPTION
**Pull request by Dianna**
Adds the new designs requested by OMSI for V5.
- replaces the taskMap with simple choose task UI
- adds modals: help others intro, chosen task description, successful task completion, unsuccessful task timeout
- a very large refactor that centralizes most of the phase controlling into the tasks redux state that had previously been in the `tasks/index.js` and `tasks/taskscreen.js`. The latter no longer needs to exist.

### CODE REVIEW

**Comment by jaronheard** [comment]
I've done some play through testing. Things seems to be working for me.

A couple of comments:
• Modal dialog timing seems too long in comparison to gameplay time
• If you select tasks where you don't have the correct item in your kit, the "Keep Playing" modal will pop up.
•  Font spacing on "You Saved" and "Badges Earned" in title bar is off
image
* When the thunderstorm is displayed, it says "It's cold out here", but blankets aren't useful, which is confusing. Should say something like "It's wet out here".

image

File: packages/2019-disaster-game/src/components/Game/KitScreen/index.js [hunk 1]
```diff
@@ -15,10 +15,10 @@ import {
 } from "../../../state/chapters";
 import {
   getKitCreationItems,
-  addItemToPlayerKit,
-  getPlayerKit
+  getPlayerKit,
+  addItemToPlayerKit
 } from "../../../state/kit";
-import { addPoints, addBadge } from "../../../state/user";
+import { earnBadge as _earnBadge, getBadges } from "../../../state/tasks";
```
Comment (jaronheard) [inline packages/2019-disaster-game/src/components/Game/KitScreen/index.js]: There's a mix of syntax used for actionCreators are referred to in this file and it's a bit confusing. Some have the _actionCreator prefix, and some don't. Should probably be consistent.

For the future, interested in what the idiomatic / most common way is.

File: packages/2019-disaster-game/src/components/Game/KitScreen/index.js [hunk 2]
```diff
@@ -65,14 +65,14 @@ const bg3 = css`
...
-  addPreparerBadge,
-  addPointsToState,
-  addItemToPlayerKitInState,
   endChapter,
   chapterDuration,
   restartGame,
   playAudio,
-  stopAudio
+  stopAudio,
+  addItemToPlayerKitInState,
```
Comment (jaronheard) [inline packages/2019-disaster-game/src/components/Game/KitScreen/index.js]: should this be `addItemToPlayerKit`?

File: packages/2019-disaster-game/src/components/Game/TaskScreen/SolveScreen.js [hunk 3]
```diff
@@ -72,7 +67,7 @@ const SolveScreen = ({
         <RequiredItemsOrbs
           css={requiredItemsStyle}
           numberRequiredItems={activeTask.numberItemsToSolve}
-          correctItemsChosen={correctItemsChosen}
+          correctItemsChosen={activeTask.numberCorrectChosen} // TODO
```
Comment (jaronheard) [inline packages/2019-disaster-game/src/components/Game/TaskScreen/SolveScreen.js]: What's the TODO here?

File: packages/2019-disaster-game/src/components/atoms/NewBadge.js [hunk 4]
```diff
@@ -59,70 +54,68 @@ const titleText = css`
   font-size: 14rem;
 `;
 
-const NewBadge = ({ type, badges, playAudio, stopAudio }) => {
-  const [badgeInfo, setBadgeInfo] = useState(badges[type]);
+const NewBadge = ({
+  badgeData,
+  displayBadgeTime = 11,
```
Comment (jaronheard) [inline packages/2019-disaster-game/src/components/atoms/NewBadge.js]: Should this timing be pulled out to live with other timing variables?

File: packages/2019-disaster-game/src/components/atoms/NewBadge.js [hunk 5]
```diff
@@ -59,70 +54,68 @@ const titleText = css`
...
+  playAudio,
+  stopAudio,
+  finishDisplayCallback,
+  fade = true
+}) => {
+  const [badgeInfo, setBadgeInfo] = useState(badgeData);
   const [hideBadge, setHideBadge] = useState(false);
+  const fadeBadgeTimeout = (displayBadgeTime - 1) * 1000;
```
Comment (jaronheard) [inline packages/2019-disaster-game/src/components/atoms/NewBadge.js]: Should this timing be pulled out to live with other timing variables?

File: packages/2019-disaster-game/src/components/atoms/TitleBar/SavedBar.js [hunk 6]
```diff
@@ -3,16 +3,9 @@ import { css, jsx } from "@emotion/core";
 import { useState, useEffect } from "react";
 import PropTypes from "prop-types";
 import { connect } from "react-redux";
-import { bindActionCreators } from "redux";
 
-import { getSavedMetrics } from "../../../state/user";
-import usePrevious from "../../../state/hooks/usePrevious";
+import { getSavedMetrics as _getSavedMetrics } from "../../../state/tasks";
```
Comment (jaronheard) [inline packages/2019-disaster-game/src/components/atoms/TitleBar/SavedBar.js]: syntax for actionCreators?

File: packages/2019-disaster-game/src/components/Game/KitScreen/index.js (hunk 1 above)
Comment (Dianna) [reply packages/2019-disaster-game/src/components/Game/KitScreen/index.js]: Updated all action creators to follow _actionCreator syntax

File: packages/2019-disaster-game/src/components/Game/KitScreen/index.js (hunk 2 above)
Comment (Dianna) [reply packages/2019-disaster-game/src/components/Game/KitScreen/index.js]: resolved by updating action creator syntax

File: packages/2019-disaster-game/src/components/Game/TaskScreen/SolveScreen.js (hunk 3 above)
Comment (Dianna) [reply packages/2019-disaster-game/src/components/Game/TaskScreen/SolveScreen.js]: Ain't one. Removed

File: packages/2019-disaster-game/src/components/atoms/NewBadge.js (hunk 4 above)
Comment (Dianna) [reply packages/2019-disaster-game/src/components/atoms/NewBadge.js]: I think for a default value it's ok?

File: packages/2019-disaster-game/src/components/atoms/NewBadge.js (hunk 5 above)
Comment (Dianna) [reply packages/2019-disaster-game/src/components/atoms/NewBadge.js]: Which variables are you thinking of? This modal gets used across a couple of chapters

File: packages/2019-disaster-game/src/components/atoms/TitleBar/SavedBar.js (hunk 6 above)
Comment (Dianna) [reply packages/2019-disaster-game/src/components/atoms/TitleBar/SavedBar.js]: updated

**Comment by Dianna** [comment]
Responding to your comments:

- For the modal timing: I got comments from them before that they wanted them to be 2x as long… I think I should leave them as is and then if they think they’re too long they’re easy enough to shorten up but I agree they are rather long
- Keep Playing modal popping up during a task player doesn't have the item for. Right now this popup is set up to go off on a timer, I could make the timer restart when this case comes up if that sounds good?
- Font spacing looks correct on the big screen, but never on smaller screens set to the same resolution for some reason. In any case, the titlebar will be redone in one of the next two releases

File: packages/2019-disaster-game/src/components/atoms/NewBadge.js (hunk 5 above)
Comment (jaronheard) [reply packages/2019-disaster-game/src/components/atoms/NewBadge.js]: I guess it's getting calculated based on displayBadgeTime, which is being passed in as a prop, so it's fine

File: packages/2019-disaster-game/src/components/atoms/NewBadge.js (hunk 4 above)
Comment (jaronheard) [reply packages/2019-disaster-game/src/components/atoms/NewBadge.js]: Agreed

**Comment by jaronheard** [comment]
Sounds good!

 > Keep Playing modal popping up during a task player doesn't have the item for. Right now this popup is set up to go off on a timer, I could make the timer restart when this case comes up if that sounds good?

Maybe it should just be longer rather than resetting? If it resets and you didn't add anything relevant to your kit then would the game keep playing itself for too long? Happy with what you think is best.

**Comment by Dianna** [comment]
Woops! I made it 10s while developing. It's now set to 35s. Good catch!

**Pull request merged**